libtenzir/builtins/operators/every_cron.cpp

```diff
@@ -230,27 +288,77 @@ class every_plugin final : public virtual operator_plugin<every_operator> {
         .primary(interval_data.source)
         .throw_();
     }
-    auto result = p.parse_operator();
-    if (not result.inner) {
-      diagnostic::error("failed to parse operator")
-        .primary(result.source)
+    return every_scheduler{*interval};
+  }
+
+private:
+  duration interval_;
+};
+using every_operator = scheduled_execution_operator<every_scheduler>;
+using every_plugin = scheduled_execution_plugin<every_scheduler>;
+
+/// scheduler for the 'cron' operator
+class cron_scheduler {
+public:
+  cron_scheduler() = default;
+  explicit cron_scheduler(detail::cron::cronexpr expr)
+    : expr_{std::move(expr)} {
+  }
+
+  constexpr static std::string_view name = "cron";
+
+  auto next_after(time::clock::time_point now) const noexcept {
+    const auto tt = time::clock::to_time_t(now);
+
+    return time::clock::from_time_t(detail::cron::cron_next(expr_, tt));
+  }
+
+  auto field(auto& f) {
+    const auto get = [this]() {
+      return detail::cron::to_cronstr(expr_);
+    };
+    const auto set = [this](std::string_view text) {
+      expr_ = detail::cron::make_cron(text);
+    };
+    return f.field("cronexpr", get, set);
+  }
+
+  static cron_scheduler parse(parser_interface& p) {
+    auto cronexpr_string = p.accept_shell_arg();
+    if (not cronexpr_string) {
+      diagnostic::error("expected cron expression")
+        .primary(p.current_span())
         .throw_();
     }
-    if (auto* pipe = dynamic_cast<pipeline*>(result.inner.get())) {
-      auto ops = std::move(*pipe).unwrap();
-      for (auto& op : ops) {
-        op = std::make_unique<every_operator>(std::move(op), *interval);
-        // Only the first operator can be a source and needs to be replaced.
-        break;
+    try {
+      return cron_scheduler{detail::cron::make_cron(cronexpr_string->inner)};
+    } catch (const detail::cron::bad_cronexpr& ex) {
+      // croncpp re-throws the message from the stoul failure
+      // which happens for most cases of invalid expressions
+      // libstdc++ and libc++ contain the string "stoul" in their what()
+      // we can check for this and provide a slightly better error message
```

Comment:
We would usually use full sentences with proper capitalization and punctuation in comments.